File: goal-keeper/src/data/dto/ClubTeamDto.ts
import {JsonObject, JsonProperty} from "json2typescript";
import PlayerDto from "@/data/dto/PlayerDto";

@JsonObject('ClubTeamDto')
export default class ClubTeamDto {

    @JsonProperty('id', String)
    public id: string | undefined = undefined;

    @JsonProperty('name', String)
    public name: string | undefined = undefined;

    @JsonProperty('players', [PlayerDto])
    public players: PlayerDto[] = [];

    @JsonProperty('competition', String)
    public competition: string | undefined = undefined;

    @JsonProperty('country', String)
    public country: string | undefined = undefined;
}

File: goal-keeper-backend/src/main/java/com/goalkeeper/api/dto/ClubTeamDto.java
package com.goalkeeper.api.dto;

import com.fasterxml.jackson.annotation.JsonRootName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Set;
import java.util.UUID;

@Builder
@Data
@JsonRootName("team")
@NoArgsConstructor
@AllArgsConstructor
public class ClubTeamDto {

    private UUID id;
    private String name;
    private List<PlayerDto> players;
    private String competition;
    private String country;
}
